File: Weather-API.Backend/Services/weatherService.js
async function getWeather(city) {
  const apiKey = process.env.VISUAL_CROSSING_API_KEY;

  const url = `https://weather.visualcrossing.com/VisualCrossingWebServices/rest/services/timeline/${city}?key=${apiKey}&unitGroup=metric&lang=pt`;

  const response = await fetch(url);

  if (!response.ok) {
    throw new Error('Erro ao buscar dados da API externa');
  }

  const data = await response.json();

  if (
    data.resolvedAddress &&
    !data.resolvedAddress.toLowerCase().includes(city.toLowerCase())
    ) {
    throw new Error('Cidade não encontrada');
    }

  return {
    city: data.address,
    temperature: data.currentConditions.temp,
    description: data.currentConditions.conditions
  };
}

module.exports = { getWeather };

File: Weather-API.Backend/Services/WeatherService.cs
using System.Text.Json;

public class WeatherService : IWeatherService
{
    private readonly HttpClient _httpClient;
    private readonly IConfiguration _configuration;

    public WeatherService(HttpClient httpClient, IConfiguration configuration)
    {
        _httpClient = httpClient;
        _configuration = configuration;
    }

    public async Task<WeatherModel?> GetWeatherAsync(string city)
    {
        var apiKey = _configuration["VisualCrossing:ApiKey"];
        var baseUrl = _configuration["VisualCrossing:BaseUrl"];

        var url = $"{baseUrl}/{Uri.EscapeDataString(city)}?unitGroup=metric&key={apiKey}&contentType=json";

        var response = await _httpClient.GetAsync(url);

        if (!response.IsSuccessStatusCode)
            return null;

        var content = await response.Content.ReadAsStringAsync();

        using var json = JsonDocument.Parse(content);
        var root = json.RootElement;

        return new WeatherModel
        {
            City = root.GetProperty("address").GetString() ?? "",
            Temperature = root.GetProperty("currentConditions").GetProperty("temp").GetDouble(),
            Description = root.GetProperty("currentConditions").GetProperty("conditions").GetString() ?? ""
        };
    }
}